core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -426,6 +459,24 @@ public Endpoint withIpAddr(@Nullable String ipAddr) {
         throw new IllegalArgumentException("ipAddr: " + ipAddr + " (expected: an IP address)");
     }
 
+    /**
+     * Returns a new host endpoint with the specified {@link InetAddress}.
+     *
+     * @return the new endpoint with the specified {@link InetAddress}.
+     *         {@code this} if this endpoint has the same IP address.
+     */
+    public Endpoint withIpAddr(InetAddress address) {
```

Comment:
Isn't this a breaking change because a user cannot pass `null` as before? (requires explicit cast to `String`)

Reply:
Indeed. How about renaming this to `withInetAddress(InetAddress)`?

Reply:
Sounds good to me.